# **PROJECT 1: WOMEN'S SHOE PRICES DATA**

# Women’s Shoe Prices - Quick Exploratory Analysis

**Goal:** Exploring price levels in a scraped dataset of women’s shoes and summarize central tendency:
mean, median, and mode. Then repeating the calculations using only the Python standard library.

**Dataset:** `womenshoeprices.csv` (≈33k rows). Numeric column used: **`prices.amountMin`** (USD).

**Source:** (https://www.kaggle.com/datasets/datafiniti/womens-shoes-prices?resource=download).  
_Why this dataset?_ It’s unfamiliar, large enough, available as CSV, has a clean numeric column, fun, and takes my interest.

**Notes & caveats:**
- Brand is missing for some rows (`nan`); I call that out where relevant.
- Prices may include sale or list values; I use the minimum observed price per listing.
- No sensitive info or API keys are present in this notebook.


## **Step 1: Read the data (pandas)**

In [7]:
import pandas as pd

# Replace with your actual filename
df = pd.read_csv("/Users/apple/Desktop/Computing in Context/Project1/womenshoeprices.csv")

df.head()

/var/folders/xb/yxxwsh2106d6ys1scdm4px9h0000gn/T/ipykernel_84767/3073035007.py:4: DtypeWarning: Columns (13,16,18,22,24,32,33,40,41,47,51) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("/Users/apple/Desktop/Computing in Context/Project1/womenshoeprices.csv")


id asins   brand  \
0  AVpe__eOilAPnD_xSt-H   NaN  Novica   
1  AVpe__eOilAPnD_xSt-H   NaN  Novica   
2  AVpe__eOilAPnD_xSt-H   NaN  Novica   
3  AVpe__eOilAPnD_xSt-H   NaN  Novica   
4  AVpe__eOilAPnD_xSt-H   NaN  Novica   

                                categories  colors  count  \
0  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
1  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
2  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
3  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
4  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   

              dateAdded           dateUpdated  \
0  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
1  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
2  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
3  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
4  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   

                                        descriptions dimension  ...  \
0  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
1  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
2  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
3  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
4  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   

                                                skus  \
0  [{"sourceURLs":["https://www.overstock.com/Wor...   
1  [{"sourceURLs":["https://www.overstock.com/Wor...   
2  [{"sourceURLs":["https://www.overstock.com/Wor...   
3  [{"sourceURLs":["https://www.overstock.com/Wor...   
4  [{"sourceURLs":["https://www.overstock.com/Wor...   

                                          sourceURLs  upc vin  websiteIDs  \
0  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
1  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
2  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
3  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
4  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   

  weight Unnamed: 48 Unnamed: 49 Unnamed: 50 Unnamed: 51  
0    NaN         NaN         NaN         NaN         NaN  
1    NaN         NaN         NaN         NaN         NaN  
2    NaN         NaN         NaN         NaN         NaN  
3    NaN         NaN         NaN         NaN         NaN  
4    NaN         NaN         NaN         NaN         NaN  

[5 rows x 52 columns]

In [3]:
df.info()
df.columns


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33801 entries, 0 to 33800
Data columns (total 52 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   33801 non-null  object 
 1   asins                2422 non-null   object 
 2   brand                32817 non-null  object 
 3   categories           33801 non-null  object 
 4   colors               19844 non-null  object 
 5   count                0 non-null      float64
 6   dateAdded            33801 non-null  object 
 7   dateUpdated          33801 non-null  object 
 8   descriptions         28425 non-null  object 
 9   dimension            1177 non-null   object 
 10  ean                  9778 non-null   float64
 11  features             27181 non-null  object 
 12  flavors              0 non-null      float64
 13  imageURLs            22309 non-null  object 
 14  isbn                 0 non-null      float64
 15  keys                 33801 non-null 

Index(['id', 'asins', 'brand', 'categories', 'colors', 'count', 'dateAdded',
       'dateUpdated', 'descriptions', 'dimension', 'ean', 'features',
       'flavors', 'imageURLs', 'isbn', 'keys', 'manufacturer',
       'manufacturerNumber', 'merchants', 'name', 'prices.amountMin',
       'prices.amountMax', 'prices.availability', 'prices.color',
       'prices.condition', 'prices.count', 'prices.currency',
       'prices.dateAdded', 'prices.dateSeen', 'prices.flavor', 'prices.isSale',
       'prices.merchant', 'prices.offer', 'prices.returnPolicy',
       'prices.shipping', 'prices.size', 'prices.source', 'prices.sourceURLs',
       'prices.warranty', 'quantities', 'reviews', 'sizes', 'skus',
       'sourceURLs', 'upc', 'vin', 'websiteIDs', 'weight', 'Unnamed: 48',
       'Unnamed: 49', 'Unnamed: 50', 'Unnamed: 51'],
      dtype='object')

## **Step 2: Mean / Median / Mode (Pandas)**

In [6]:
# Using the numeric column prices.amountMin
mean_price = df["prices.amountMin"].mean()
median_price = df["prices.amountMin"].median()
mode_price = df["prices.amountMin"].mode()[0]   # mode() returns a Series

# Put results into a small table
stats_table = pd.DataFrame({
    "Statistic": ["Mean", "Median", "Mode"],
    "Value ($)": [round(mean_price, 2), round(median_price, 2), round(mode_price, 2)]
})

stats_table

Statistic  Value ($)
0      Mean      87.66
1    Median      52.99
2      Mode      39.99

**Interpretation:** Mean reflects the average price (sensitive to outliers), the median is the “middle”
typical price, and the mode is the most common price point in the dataset.

## **Step 3: Pure Python (The Hard Way)**

In [ ]:
import csv

values = []

with open("womenshoeprices.csv", "r") as f:
    reader = csv.DictReader(f)
    for row in reader:
        val_str = row["prices.amountMin"]
        if val_str != "" and val_str is not None:
            try:
                val = float(val_str)
                values.append(val)
            except:
                pass

# Mean (hard way)
mean_hard = sum(values) / len(values)

# Median (hard way)
sorted_vals = sorted(values)
mid = len(sorted_vals) // 2

if len(sorted_vals) % 2 == 1:
    median_hard = sorted_vals[mid]
else:
    median_hard = (sorted_vals[mid - 1] + sorted_vals[mid]) / 2

# Mode (hard way)
counts = {}
for v in values:
    counts[v] = counts.get(v, 0) + 1

max_count = max(counts.values())
mode_hard = [v for v, c in counts.items() if c == max_count]

print(mean_hard)
print(median_hard)
print(mode_hard)


87.65579834916127
52.99
[39.99]


_These values match the pandas results (within rounding), confirming the manual implementation._ 
For 

## **Step 4: Visualization (standard library only)**

In [ ]:
# Showing 15 shoes with text bars for price
sample = df.sample(15)

for i, row in sample.iterrows():
    price = row["prices.amountMin"]
    if pd.notna(price):
        bar = "$" * int(price // 10)   # one $ per $10
        print(f"{row['brand']}: {bar}")


MUK LUKS A LA MODE: $
Dromedaris: $$$$$$$$$
Journee Collection: $$$$$
nan: $$$$$$
Spring Step: $$$$$$$
In-Sattva: $$$$$$$
Grasshoppers: $$$
Touch Ups: $$$$
Khombu: $$$$$$$$$
White Mountain: $$$$$$
Naturalizer: $$$$$
Soft Style: $$$$
David Tate: $$$$$$$$
SoftWalk: $$$$$$$$$$$$$
adidas: $$$$$$$


**How to read this.** Each `$` symbol represents approximately **$${DOLLARS_PER_SYMBOL}**.
More `$` = higher price. 

This text chart conveys relative prices without any plotting libraries.

### Summary & Takeaways

For this project, I worked with a dataset of **33,801 women’s shoe listings**.  
The numeric column I analyzed was **`prices.amountMin`**, which represents the minimum observed price for each item.

First, I used **pandas** to calculate the **mean, median, and mode** of shoe prices.  
Then, following the project requirements, I repeated the same calculations using **only the standard Python library** specifically `csv`, lists, loops, and dictionaries without relying on pandas or any statistical shortcuts.

To visualize the results, I built a simple **text-based price chart** using `$` symbols.  
This works similarly to how Google Maps uses `$`, `$$`, or `$$$` to indicate how expensive a restaurant is.  
Since shoe prices vary more than restaurant meals, the number of `$` symbols can be much higher.  
In my chart, **each `$` represents about $10**, making it easy to compare relative prices at a glance, without needing any external plotting libraries.

Overall, this project gave me practice with:
- Working with a large dataset in CSV format
- Computing summary statistics in pandas **and** “the hard way”
- Creating a visualization using only the standard Python library
